unity-renderer/Assets/DCLPlugins/ECS7/Systems/VideoPlayerSystem/Tests/ECSVideoPlayerShould.cs

```diff
@@ -0,0 +1,109 @@
+
+            videoPlayer = new WebVideoPlayer("test", "test.mp4", true, new VideoPluginWrapper_Mock());
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            testUtils?.Dispose();
+            videoPlayer?.Dispose();
+            AssetPromiseKeeper_Material.i.Cleanup();
+            AssetPromiseKeeper_Texture.i.Cleanup();
+        }
+
+        [UnityTest]
+        public IEnumerator ApplyVideoMaterial()
+        {
+            var playerComponent = internalEcsComponents.videoPlayerComponent;
+            var videoMaterialComponent = internalEcsComponents.videoMaterialComponent;
+
+            ECS7TestEntity entity00 = scene0.CreateEntity(100);
+            ECS7TestEntity entity01 = scene0.CreateEntity(101);
+            ECS7TestEntity entity10 = scene1.CreateEntity(200);
+
+            playerComponent.PutFor(scene0, entity00, new InternalVideoPlayer()
+            {
+                videoPlayer = videoPlayer,
+                assignedMaterials = new List<InternalVideoPlayer.MaterialAssigned>(),
+            });
+
+            videoMaterialComponent.PutFor(scene0, entity00, new InternalVideoMaterial()
+            {
+                material = new Material(Shader.Find("Standard")) { mainTexture = new Texture2D(1, 1), },
+                videoTextureDatas = new List<InternalVideoMaterial.VideoTextureData>() { new InternalVideoMaterial.VideoTextureData(100, ShaderUtils.BaseMap) },
+            });
+
+            videoMaterialComponent.PutFor(scene0, entity01, new InternalVideoMaterial()
+            {
+                material = new Material(Shader.Find("Standard")) { mainTexture = new Texture2D(1, 1), },
+                videoTextureDatas = new List<InternalVideoMaterial.VideoTextureData>() { new InternalVideoMaterial.VideoTextureData(100, ShaderUtils.BaseMap) },
+            });
+
+            // Texture should not change on this
+            videoMaterialComponent.PutFor(scene1, entity10, new InternalVideoMaterial()
+            {
+                material = new Material(Shader.Find("Standard")) { mainTexture = new Texture2D(1, 1), },
+                videoTextureDatas = new List<InternalVideoMaterial.VideoTextureData>() { new InternalVideoMaterial.VideoTextureData(200, ShaderUtils.BaseMap) },
+            });
+
+            yield return new WaitUntil(() => videoPlayer.GetState() == VideoState.READY);
+            videoPlayer.Update();
+            Assert.IsNotNull(videoPlayer.texture);
+
+            systemsUpdate();
+
+            var videoPlayerComponent = playerComponent.GetFor(scene0, entity00);
+            var videoMaterial00 = videoMaterialComponent.GetFor(scene0, entity00);
+            var videoMaterial01 = videoMaterialComponent.GetFor(scene0, entity01);
+            var videoMaterial10 = videoMaterialComponent.GetFor(scene1, entity10);
+
+            Assert.True(ReferenceEquals(videoMaterial00.model.material.GetTexture(ShaderUtils.BaseMap), videoPlayerComponent.model.videoPlayer.texture));
```

Comment:
`_BaseMap` property is going to work only if you use `Universal Render Pipeline/Lit`. 
The texture property of `Standard` is `_MainTex`